mne/channels/interpolation.py

```diff
@@ -251,25 +273,94 @@ def _interpolate_bads_nirs(inst, method="nearest", exclude=(), verbose=None):
-            # Ignore distances to other bad channels
-            dists_to_bad[bads_mask] = np.inf
-            # Find closest remaining channels for same frequency
-            closest_idx = np.argmin(dists_to_bad) + (bad % 2)
-            inst._data[bad] = inst._data[closest_idx]
-    else:
-        assert method == "nan"
-        inst._data[picks_bad] = np.nan
+    dist = pdist(locs3d)
+    dist = squareform(dist)
+
+    for bad in picks_bad:
+        dists_to_bad = dist[bad]
+        # Ignore distances to self
+        dists_to_bad[dists_to_bad == 0] = np.inf
+        # Ignore distances to other bad channels
+        dists_to_bad[bads_mask] = np.inf
+        # Find closest remaining channels for same frequency
+        closest_idx = np.argmin(dists_to_bad) + (bad % 2)
+        inst._data[bad] = inst._data[closest_idx]
+
     # TODO: this seems like a bug because it does not respect reset_bads
     inst.info["bads"] = [ch for ch in inst.info["bads"] if ch in exclude]
 
     return inst
+
+
+@verbose
+def _interpolate_bads_seeg(inst, exclude=None, tol=2e-3, verbose=None):
+    if exclude is None:
+        exclude = list()
+    bads_idx = np.zeros(len(inst.ch_names), dtype=bool)
+    goods_idx = np.zeros(len(inst.ch_names), dtype=bool)
+
+    picks = pick_types(inst.info, meg=False, seeg=True, exclude=exclude)
+    inst.info._check_consistency()
+    bads_idx[picks] = [inst.ch_names[ch] in inst.info["bads"] for ch in picks]
+
+    if len(picks) < 3 or bads_idx.sum() == 0:
+        return
+
+    goods_idx[picks] = True
+    goods_idx[bads_idx] = False
+
+    pos = inst._get_channel_positions(picks)
+
+    # Make sure only sEEG are used
+    bads_idx_pos = bads_idx[picks]
+
+    # for each bad contact:
+    # 1) find nearest neighbor to define the electrode shaft line
+    # 2) find all contacts on the same line
+    # 3) interpolate the bad contacts
+
+    dist = squareform(pdist(pos))
+    np.fill_diagonal(dist, np.inf)
+
+    picks_bad = list(np.where(bads_idx_pos)[0])
+    while picks_bad:
+        bad = picks_bad[0]
+        n1 = pos[bad]
+        n2 = pos[np.argmin(dist[bad])]  # 1
+        # https://mathworld.wolfram.com/Point-LineDistance3-Dimensional.html
+        shaft_dists = np.array(
+            [
+                0
+                if all(n0 == n1) or all(n0 == n2)
+                else np.linalg.norm(np.cross((n0 - n1), (n0 - n2)))
+                / np.linalg.norm(n2 - n1)
+                for n0 in pos
+            ]
+        )
+        shaft = np.where(shaft_dists < tol)[0]  # 2
+        if shaft.size < 3:
+            raise RuntimeError(
+                f"Only {shaft.size} contact positions in a line "
+                f" found for {inst.ch_names[bad]}, 3 required for "
+                "interpolation, fix the positions or exclude this channel"
+            )
+        bads_shaft = np.array([idx for idx in picks_bad if idx in shaft])
+        goods_shaft = shaft[~np.isin(shaft, bads_shaft)]
```

Comment:
[nitpick] might be slightly faster to leverage `np.isin(..., invert=True)`?  To me it's also slightly more readable (the `~` negation is a bit confusible with `-` for me on quick glances)